RCS file: /cvs/gsl/vcl/source/window/toolbox.cxx,v
retrieving revision 1.9.4.2
diff -u -r1.9.4.2 toolbox.cxx
--- vcl/source/window/toolbox.cxx	24 May 2002 14:10:50 -0000	1.9.4.2
+++ vcl/source/window/toolbox.cxx	22 Nov 2002 17:49:23 -0000
@@ -2566,27 +2566,29 @@
             pTempItem = mpItemList->GetObject( nPos+1 );
             if ( pTempItem && !pTempItem->mbShowWindow && !pTempItem->mbBreak )
             {
-                long nCenterPos;
+                long nCenterPos, nSlim;
                 SetLineColor( rStyleSettings.GetShadowColor() );
                 if ( IsHorizontal() )
                 {
+		    nSlim = (pItem->maRect.Bottom() - pItem->maRect.Top ()) / 4;
                     nCenterPos = pItem->maRect.Center().X()-1;
-                    DrawLine( Point( nCenterPos, pItem->maRect.Top() ),
-                              Point( nCenterPos, pItem->maRect.Bottom() ) );
+                    DrawLine( Point( nCenterPos, pItem->maRect.Top() + nSlim - 1),
+                              Point( nCenterPos, pItem->maRect.Bottom() - nSlim - 1) );
                     nCenterPos++;
                     SetLineColor( rStyleSettings.GetLightColor() );
-                    DrawLine( Point( nCenterPos, pItem->maRect.Top() ),
-                              Point( nCenterPos, pItem->maRect.Bottom() ) );
+                    DrawLine( Point( nCenterPos, pItem->maRect.Top() + nSlim),
+                              Point( nCenterPos, pItem->maRect.Bottom() - nSlim) );
                 }
                 else
                 {
+		    nSlim = (pItem->maRect.Right() - pItem->maRect.Left ()) / 4;
                     nCenterPos = pItem->maRect.Center().Y()-1;
-                    DrawLine( Point( pItem->maRect.Left(), nCenterPos ),
-                              Point( pItem->maRect.Right(), nCenterPos ) );
+                    DrawLine( Point( pItem->maRect.Left() + nSlim, nCenterPos),
+                              Point( pItem->maRect.Right() - nSlim, nCenterPos) );
                     nCenterPos++;
                     SetLineColor( rStyleSettings.GetLightColor() );
-                    DrawLine( Point( pItem->maRect.Left(), nCenterPos ),
-                              Point( pItem->maRect.Right(), nCenterPos ) );
+                    DrawLine( Point( pItem->maRect.Left() + nSlim, nCenterPos ),
+                              Point( pItem->maRect.Right() - nSlim, nCenterPos ) );
                 }
             }
         }
@@ -2865,36 +2868,26 @@
     // Evt. noch Pfeil rechts/oben in der Ecke zeichnen
     if ( pItem->mnBits & TIB_DROPDOWN )
     {
+#define ARROW_SIZE_HIGHLIGHT 8
+#define ARROW_SIZE_NO_HIGHLIGHT 6
         Point aArrowPos( nOffX, nOffY );
-        aArrowPos.X() += nBtnWidth-6;
+	int arrow_size = bHighlight ? ARROW_SIZE_HIGHLIGHT : ARROW_SIZE_NO_HIGHLIGHT;
+        aArrowPos.X() += nBtnWidth - arrow_size;
 
         Color       aOldLineColor = GetLineColor();
         Color       aOldFillColor = GetFillColor();
-        Rectangle   aClearRect( aArrowPos.X()-1, aArrowPos.Y(),
-                                aArrowPos.X()+3, aArrowPos.Y()+4 );
         SetLineColor();
-
+#if 0
         if ( (meAlign == WINDOWALIGN_LEFT) || (meAlign == WINDOWALIGN_RIGHT) )
-        {
-            aArrowPos.X()       += 2;
-            aClearRect.Left()   += 2;
-            aClearRect.Right()  += 2;
-            aClearRect.Bottom() += 2;
-        }
-        else
-            aClearRect.Right()  += 2;
-
-        Erase( aClearRect );
-        BOOL bBlack = FALSE;
-
+            aArrowPos.X() += 2;
+#endif
         if ( !pItem->mbEnabled || !IsEnabled() )
-            SetFillColor( rStyleSettings.GetShadowColor() );
+            SetFillColor( rStyleSettings.GetShadowColor () );
         else
-        {
-            SetFillColor( COL_LIGHTGREEN );
-            bBlack = TRUE;
-        }
-        ImplDrawToolArrow( this, aArrowPos.X(), aArrowPos.Y(), bBlack );
+	    SetFillColor( rStyleSettings.GetMenuHighlightColor () );
+
+	ImplDrawToolArrow( this, aArrowPos.X(), aArrowPos.Y(),
+			   TRUE, FALSE, FALSE, arrow_size );
         SetLineColor( aOldLineColor );
         SetFillColor( aOldFillColor );
     }
